### PR TITLE
BUG: `highs-ds` returns memoryviews instead of np.arrays for the dual values

### LINKED ISSUE #15339: BUG: `highs-ds` returns memoryviews instead of np.arrays for the dual values
According to the docs, the `highs-ds` method returns the Lagrangian multipliers (also known as *dual values*) corresponding to both the inequality/equality constraints and the upper/lower bounds as np.ndarray. However, for the latter it returns memoryviews.

# Reproducing Code Example

``` python
import numpy as np
from scipy.optimize import linprog

c = -1*np.array([300, 500])
A_ub = np.array([[1, 2], [1, 1], [0, 3]])
b_ub = np.array([170, 150, 180])
bounds = [(0, 100), (0, 100)]

# solve c'x  s.t.  A_ub*x <= b_ub,  0 <= x <= 100
result = linprog(c=c, A_ub=A_ub, b_ub=b_ub, bounds=bounds, method="highs-ds")
for key in ['lower', 'upper', 'eqlin', 'ineqlin']:
    print(result[key]['marginals'])
```

yields

``` none
<MemoryView of 'ndarray' object>
<MemoryView of 'ndarray' object>
[]
[-250.   -0.   -0.]
```

---

I had a look at the Cython wrapper and it indeed returns a memoryview which isn't converted to np.ndarray:

https://github.com/scipy/scipy/blob/a0e129fc3ba89fd81a5fa7e5e4d6aa2ec635a674/scipy/optimize/_highs/cython/src/_highs_wrapper.pyx#L706

https://github.com/scipy/scipy/blob/a0e129fc3ba89fd81a5fa7e5e4d6aa2ec635a674/scipy/optimize/_linprog_highs.py#L385-L386

### PR DESCRIPTION
**Pull request by jhelgert**
Closes #15339.

### CODE REVIEW

**Comment by mdhaber** [comment]
Thanks. Would you add `assert isinstance` in `test_optimize_result` in `scipy/optimize/tests/test_linprog.py` to check that the objects are arrays (if they are present)? You can use regular `assert`s; no need to use the `assert_` that you see there already.

File: scipy/optimize/tests/test_linprog.py [hunk 1]
```diff
@@ -1172,6 +1172,10 @@ def test_optimize_result(self):
         assert_allclose(c @ res.x, res.fun)
         assert_allclose(b_eq - A_eq @ res.x, res.con, atol=1e-11)
         assert_allclose(b_ub - A_ub @ res.x, res.slack, atol=1e-11)
+        for key in ['eqlin', 'ineqlin', 'lower', 'upper']:
+            if key in res.keys():
+                assert isinstance(res[key]['marginals'], np.ndarray)
+                assert isinstance(res[key]['residual'], np.ndarray)
```
Comment (tylerjereddy) [inline scipy/optimize/tests/test_linprog.py:1178]: This indeed looks like what Matt was asking for--I'll let them confirm.

Comment (mdhaber) [reply scipy/optimize/tests/test_linprog.py:1178]: Yes, I was just going to test it locally and then merge. @mckib2 look good to you, too?

Comment (mckib2) [reply scipy/optimize/tests/test_linprog.py:1178]: Yep, merge away!

**Review by mckib2 — APPROVED** [review]

**Comment by mdhaber** [comment]
Thanks @jhelgert @tylerjereddy @mckib2! Please do backport this one if possible, @tylerjereddy.

**Comment by tylerjereddy** [comment]
yep, will do

**Pull request merged**